src/player_com.rs

```diff
@@ -0,0 +1,167 @@
+use std::fmt;
+use std::io::{BufRead, BufReader, Write};
+use std::process::{Command, Stdio};
+use std::sync::mpsc;
+use std::sync::mpsc::{Receiver, RecvTimeoutError, Sender};
+use std::thread;
+use std::time::Duration;
+
+use super::plateau::Player;
+
+pub struct PlayerError {
+    player: Player,
+    msg: String,
+}
+
+impl PlayerError {
+    pub fn new(player: Player, msg: String) -> Self {
+        Self { player, msg }
+    }
+}
+
+impl fmt::Display for PlayerError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let num = match self.player {
+            Player::Player1 => 1,
+            Player::Player2 => 2,
+        };
+        write!(f, "Player {} - Error: {}", num, self.msg)
+    }
+}
+
+pub struct ComError {
+    msg: String,
+}
+
+impl ComError {
+    pub fn new(msg: String) -> Self {
+        Self { msg }
+    }
+}
+
+impl fmt::Display for ComError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+pub struct PlayerCom {
+    player1_sender: Sender<std::string::String>,
+    player1_receiver: Receiver<std::string::String>,
+    player2_sender: Sender<std::string::String>,
+    player2_receiver: Receiver<std::string::String>,
+    timeout: u64,
+}
+
+impl PlayerCom {
+    pub fn new(path1: String, path2: String, timeout: u64) -> Result<PlayerCom, ComError> {
+        let (p1_sender, p1_receiver) = PlayerCom::spawn_child_process(path1, Player::Player1)?;
+        let (p2_sender, p2_receiver) = PlayerCom::spawn_child_process(path2, Player::Player2)?;
+
+        Ok(PlayerCom {
+            player1_sender: p1_sender,
+            player1_receiver: p1_receiver,
+            player2_sender: p2_sender,
+            player2_receiver: p2_receiver,
+            timeout,
+        })
+    }
+
+    pub fn p1_send(&self, message: String) -> Result<(), PlayerError> {
+        match self.player1_sender.send(message) {
+            Ok(_) => Ok(()),
+            Err(_) => Err(PlayerError::new(
+                Player::Player1,
+                String::from("Error while sending message"),
+            )),
+        }
+    }
+
+    pub fn p2_send(&self, message: String) -> Result<(), PlayerError> {
+        match self.player2_sender.send(message) {
+            Ok(_) => Ok(()),
+            Err(_) => Err(PlayerError::new(
+                Player::Player2,
+                String::from("Error while sending message"),
+            )),
+        }
+    }
+
+    pub fn p1_receive(&self) -> Result<String, PlayerError> {
+        let s = self
+            .player1_receiver
+            .recv_timeout(Duration::from_secs(self.timeout));
```

Comment:
Do you think we would ever want anything more precise than a second on the timeout, like half a second?